### DISCUSSION TITLE
Discussion: Missing classes and types in Microsoft.Foundation

### DISCUSSION THREAD

**Discussion by sjb-sjb** [microsoft/WindowsAppSDK#400] [Ideas]
Currently many of the WinUI classes and methods depend directly on types from Windows.Foundation. An example would be TypedEventHandler, and I am there are many other examples. 

It would be more logical to eliminate this dependency on Windows by putting the types into Microsoft.Foundation. After all, the objective is to separate the UI classes from Windows. 

Is it planned to make that change, or to just keep using Windows.Foundation namespace in WinUI ? If the latter then what other Windows namespaces are going to remain embedded in WinUI ? 

sjb

**Comment by huoyaoyuan** [comment]
> It would be more logical to eliminate this dependency on Windows by putting the types into Microsoft.Foundation. After all, the objective is to separate the UI classes from Windows

This should be non-goal. The UI always dependents on Windows. The goal is to dependent on **a fixed version of Windows** with different versions of UI.

**Comment by andrewleader** [comment]
I believe our plan is indeed to Reunionize these Foundation APIs to remove all direct dependencies on the platform, @anawishnoff do you have some more perspectives from the WinUI team on this topic?

**Reply by anawishnoff** [reply]
Maybe @marb2000 or @MikeHillberg might have info?
